### DISCUSSION TITLE
Dynamic MaxY and MinY values when scrolling while chart is zoomed

### DISCUSSION THREAD

**Discussion by kuti-JP** [patrykandpatrick/vico#919] [Questions]
### Question

Hi, I'm not sure if this is supported currently by Vico but is there a way to have the visible chart frame when scrolling to have it's Y-axis max and min to take up the entire screen and have this changed dynamically when you're scrolling. This allows for situations where only a piece of the entire screen is used when the maxY of the entire chart is way bigger than the current maxY value when scrolling. 

Currently I have created a custom `AxisValueOverrider` that works when I manually change `Zoomstate` and scroll using `ScrollState.scroll`, this is possible because at that point I know what range of value will be shown but not when I'm scrolling by dragging the chart.

### Vico version(s)

2.0.0-alpha.26

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, @kuti-JP.

> Hi, I'm not sure if this is supported currently by Vico but is there a way to have the visible chart frame when scrolling to have it's Y-axis max and min to take up the entire screen and have this changed dynamically when you're scrolling.

It's not supported at the moment, but it's a planned addition. See #403. I'm closing this as a duplicate.

> Currently I have created a custom `AxisValueOverrider` that works when I manually change `Zoomstate` and scroll using `ScrollState.scroll`, this is possible because at that point I know what range of value will be shown but not when I'm scrolling by dragging the chart.

You should still be able to do this when the chart is interacted with via gesture, as `ScrollState` and `ZoomState` expose the scroll and zoom values. You'd have to calculate the visible _x_ range based on these values and the information available in `CartesianDrawingContext`. In the future, `VicoScrollState` will directly expose the visible _x_ range. See #788.
